src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    public static final int FOR_GET_SALARY = 1;
+    public static final int FOR_RECORD = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, DATE_FORMAT);
+        LocalDate localDateTo = LocalDate.parse(dateTo, DATE_FORMAT);
+        StringBuilder builder = new StringBuilder("Report for period "
+                                                  + dateFrom + " - " + dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (String datum : data) {
+                String[] record = datum.split(" ");
+                LocalDate changingDate = LocalDate.parse(record[FOR_RECORD], DATE_FORMAT);
+                if (changingDate.isAfter(localDateFrom)
+                        && (changingDate.isBefore(localDateTo)
+                        || changingDate.isEqual(localDateTo))
+                        && datum.contains(name)) {
```

Comment:
we have to check for a complete match
```suggestion
                        && record[NAME].equals(name)) {
```